sc25519.c

```diff
@@ -166,7 +166,7 @@ int sc25519_lt_vartime(const sc25519 *x, const sc25519 *y)
 
 void sc25519_add(sc25519 *r, const sc25519 *x, const sc25519 *y)
 {
-  int i, carry;
+  crypto_uint32 i, carry;
```

Comment:
What warning is this fixing?

This code is taken entirely from its upstream (in supercop) and is only ever used for small integer loops so we'd need a really good reason to want to change it.